src/pages/puzzles/[instanceId]/index.tsx

```diff
@@ -38,42 +41,63 @@ const PuzzlePage = ({
     user
   ) => {
     if (isAuthenticated) {
-      const submission = await submitPuzzleInstance(
-        answer,
-        puzzleInstance,
-        puzzle,
-        randomSeed,
-        user
-      );
+      const submissionInstance = async () => {
+        const submission = await submitPuzzleInstance(
+          answer,
+          puzzleInstance,
+          puzzle,
+          randomSeed,
+          user
+        );
 
-      if ((submission as HandledError).error) {
-        // TODO: handle errors from submitting
-        alert(submission.message);
-        return;
-      }
+        if ((submission as HandledError).error) {
+          console.error((submission as HandledError).message);
+          throw new Error('Something went wrong :(');
+        } else {
+          const success = submission.isCorrect[submission.isCorrect.length - 1];
 
-      const success = submission.isCorrect.at(-1);
+          const feedbackGifs = success
+            ? allFeedbackGifs.correct
+            : allFeedbackGifs.incorrect;
 
-      const correctnessMessage = success
-        ? 'Correct! Nice work!'
-        : 'Almost there! Try Again.';
-      alert(`${correctnessMessage}`);
+          setIsRecentCorrect(success);
+          setFeedbackGifSrc(getRandomGifSrc(feedbackGifs, success));
 
-      const feedbackGifs = success
-        ? allFeedbackGifs.correct
-        : allFeedbackGifs.incorrect;
+          if (!success) {
+            throw new Error('Almost there! Try Again');
+          }
+        }
+      };
 
-      setIsRecentCorrect(success);
-      setFeedbackGifSrc(getRandomGifSrc(feedbackGifs, success));
+      toast.promise(submissionInstance(), {
+        loading: 'Submitting your answer... ⏳',
+        success: 'Correct! Nice work!',
```

Comment:
It seems weird to throw an error for incorrect answers - isn't there a way to grab whatever the promise returns and check `isCorrect`?

Reply:
I did try to use the hooks approach but somehow toast does not get the update from `useState`.
Normally, I will have `const [errMsg, setErrMsg] = useState<string>();` and then change it according to the cases. But for toast, it doesn't get the update from `setErrMsg` and will fail to send the message. So I think of this solution.
If you think of a better way, please tell me, I know this is an awful implementation 🥺

Reply:
But does this work? Did you test that it does what you think it does in every case?